### PR TITLE
Fix #9003 Make DrawerHeader margin configurable

### LINKED ISSUE #9003: Make DrawerHeader margin configurable
The Material spec says that drawer headers are supposed to scroll with the rest of the drawer.

https://material.io/guidelines/patterns/navigation-drawer.html#navigation-drawer-content

However, some apps might want a drawer header that doesn't scroll with the drawer items. There will be 8 pixels of white margin below the drawer header in this case.

We should make the drawer header margin configurable to allow overriding this.

### PR DESCRIPTION
**Pull request by collinjackson**

### CODE REVIEW

File: packages/flutter/lib/src/material/user_accounts_drawer_header.dart [hunk 1]
```diff
@@ -146,6 +147,11 @@ class UserAccountsDrawerHeader extends StatefulWidget {
   /// with its background color set to the current theme's primaryColor is used.
   final Decoration decoration;
 
+  /// The margin around the drawer header.
+  ///
+  /// If not specified, the [DrawerHeader] default is used.
```
Comment (collinjackson) [inline packages/flutter/lib/src/material/user_accounts_drawer_header.dart]: I think this is not quite right, if not specified a margin of null (0) is used. I will fix and write a test.

**Comment by HansMuller** [comment]
LGTM. As you say, specifying margin: null for either drawer header should mean: no margin and not specifying a margin should mean const EdgeInsets.only(bottom: 8.0).

File: packages/flutter/lib/src/material/drawer_header.dart [hunk 2]
```diff
@@ -55,6 +56,9 @@ class DrawerHeader extends StatelessWidget {
   /// If the child is null, the padding has no effect.
   final EdgeInsets padding;
 
+  /// The margin used to surround the child.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/drawer_header.dart]: This should discuss how it's distinct from `padding` (which has a very similar description).

**Comment by Hixie** [comment]
LGTM with tests and the nits above.

**Pull request merged**